pbxplore/tests/test_functions.py

```diff
@@ -252,7 +252,7 @@ def test_get_phi_psi_angles(self, chain, resid, angles):
         Tests for get_phi_psi_angles()
         """
         phi_psi = chain.get_phi_psi_angles()
-        assert angles["phi"] == pytest.approx(phi_psi[resid]["phi"])
+        assert (angles["phi"] is None and phi_psi[resid]["phi"] is None) or angles["phi"] == pytest.approx(phi_psi[resid]["phi"])
```

Comment:
What does mandate this change? The input did not change and the test used to pass. Did the behaviour of `pytest.approx` change? It would be reasonable.

Reply:
I confirm this test still passed with Python >=3.6 without this modification.

Reply:
I had issues with the unit test of this function, because `pytest.approx` was not accepting NaN comparisons, the behavior changed, you are still able to make it work @HubLot ?

Reply:
Yes, I was able to make it work on the version of the master with Python 3.6 to 3.8 with pytest 6.2.2
Maybe it was a bug in `pytest.approx` ?

Reply:
Interesting, I run pytest 6.2.2 also, with python 3.8.5 and I got:
```
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[11-angles0] FAILED                                                [ 33%]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[12-angles1] PASSED                                                [ 35%]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[13-angles2] FAILED                                                [ 36%]

```

Reply:
Can you give the command and the first output lines of the pytest ?

Here's mine:

```bash
(PBxplore) [hub@Serenity:PBxplore (master=)]$ pytest
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/hub/.local/share/virtualenvs/PBxplore-MAEcPXWb/bin/python
cachedir: .pytest_cache
rootdir: /home/hub/PBxplore, configfile: setup.cfg

[...]
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[11-angles0] PASSED                                                                                                              
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[12-angles1] PASSED                                                                                                              
pbxplore/tests/test_functions.py::TestChainClass::test_get_phi_psi_angles[13-angles2] PASSED
```

Reply:
Looks like there is a version dependency somewhere. The explicit form of the test is clearer and always works, let's go with that. I have the feeling the line is long, though; I would split it.
```suggestion
        assert (
            (angles["phi"] is None and phi_psi[resid]["phi"] is None)
            or angles["phi"] == pytest.approx(phi_psi[resid]["phi"])
        )
```

Reply:
Ah yes that's it, for some reason even after upgrading all my pytest versions it was still using pytest-5.4.3 in the venv.
v6.2.2 works so it was indeed a bug in pytest version. Thanks for pointing it out @HubLot !
Still want to keep the new line or should I revert as proposes @jbarnoud ?